File: Easy/2566_Maximum_Difference_by_Remapping_a_Digit/Solution.py



class Solution:
    def minMaxDifference(self, num: int) -> int:
        fast = ""
        maxNum = ""
        minNum = ""
        s = str(num)

        for i in s:
            if (i != "9"):
                fast = i
                break
        
        for i in s:
            if (i == fast):
                maxNum += "9"
            else:
                maxNum += i
        
        for i in s:
            if (i != "0"):
                fast = i
                break

        for i in s:
            if (i == fast):
                minNum += "0"
            else:
                minNum += i
             
        return int(maxNum) - int(minNum)


if __name__ == "__main__":
    solution = Solution()

    
    # test cases 1
    print(solution.minMaxDifference(11891)) # 99009

    # test cases 2
    print(solution.minMaxDifference(90)) # 99

File: Easy/2566_Maximum_Difference_by_Remapping_a_Digit/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    int minMaxDifference(int num) {
        string s = to_string(num);
        string maxNum = "";
        string minNum = "";
        char fastNum = '0';
        int n = s.size();

        for (int i = 0; i < n; i++) {
            if (s[i] != '9') {
                fastNum = s[i];
                break;
            }
        }

        for (int i = 0; i < n; i++) {
            if (s[i] == fastNum) {
                maxNum += '9';
            } else {
                maxNum += s[i];
            }
        }

        for (int i = 0; i < n; i++) {
            if (s[i] != '0') {
                fastNum = s[i];
                break;
            }
        }
        
        for (int i = 0; i < n; i++) {
            if (s[i] == fastNum) {
                minNum += '0';
            } else {
                minNum += s[i];
            }
        }

        return stoi(maxNum) - stoi(minNum);
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;
    
    // test cases 1
    cout << solution.minMaxDifference(11891) << endl; // Output: 99009

    // test cases 2
    cout << solution.minMaxDifference(90) << endl; // Output: 99

    return 0;
}